addons/website_sale/data/mail_template_data.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+        <!--Email template -->
+        <record id="recovery_mail" model="mail.template">
+            <field name="name">Abandoned Cart - Send a Recovery Email</field>
+            <field name="email_from">${(object.user_id.email or '')|safe}</field>
```

Comment:
No because I think that a user created from the website sale shop does not have a parent_id by default and the email field is mandatory when proceeding the cart.